pkg/registry/helm_test.go

```diff
@@ -0,0 +1,703 @@
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, os.RemoveAll(helmDir))
+	}()
+
+	localHelmSrcDirValid := filepath.Join("testdata", "helm", "valid")
+	localHelmSrcDirInvalid := filepath.Join("testdata", "helm", "invalid")
+
+	tests := []struct {
+		name             string
+		helmSrcDir       string
+		expectedCommands []string
+		helmChartPaths   []string
+		expectedError    string
+	}{
+		{
+			name:       "Helm Dir With Valid Manifests",
+			helmSrcDir: localHelmSrcDirValid,
+			helmChartPaths: []string{
+				filepath.Join(helmDir, "apache2.tgz"), // This one is created manually while the others are pulled
+				"apache-*.tgz",
+				"metallb-*.tgz",
+			},
+			expectedCommands: []string{
+				fmt.Sprintf("helm template --skip-crds apache2 %s", filepath.Join(helmDir, "apache2.tgz")),
+				fmt.Sprintf("helm pull oci://registry-1.docker.io/bitnamicharts/apache --version 10.5.2 -d %s", helmDir),
+				fmt.Sprintf("helm template --skip-crds apache oci://registry-1.docker.io/bitnamicharts/apache --version 10.5.2 --set rbac.enabled=true,servers[0].host=example,servers[0].port=80,servers[1].host=example2,servers[1].port=22,ssl.enabled=true -f %s", filepath.Join(helmDir, "values-apache.yaml")),
+				"helm template --skip-crds metallb repo-metallb/metallb",
+				"helm repo add repo-metallb https://suse-edge.github.io/charts",
+				fmt.Sprintf("helm pull repo-metallb/metallb -d %s", helmDir),
+				"helm template --skip-crds metallb repo-metallb/metallb",
+			},
+		},
+		{
+			name:          "Helm Dir With Invalid Manifests",
+			helmSrcDir:    localHelmSrcDirInvalid,
+			expectedError: "parsing helm manifest in 'testdata/helm/invalid/invalid.yaml'",
+		},
+		{
+			name:       "No Source Dir",
+			helmSrcDir: "",
+		},
+		{
+			name:          "Invalid Source Dir",
+			helmSrcDir:    "invalid",
+			expectedError: "getting helm manifest paths: reading manifest source dir 'invalid': open invalid: no such file or directory",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			helmCommands, helmChartPaths, err := GenerateHelmCommands(test.helmSrcDir, helmDir)
+			if test.expectedError == "" {
+				assert.ElementsMatch(t, helmChartPaths, test.helmChartPaths)
+				require.NoError(t, err)
+				for _, command := range helmCommands {
+					if !strings.Contains(command, "--set") {
+						assert.Contains(t, test.expectedCommands, command)
+					} else {
+						actualFields := strings.Fields(command)
+						expectedFields := strings.Fields(test.expectedCommands[2])
+
+						if !slices.Contains(actualFields, "--set") {
+							assert.ElementsMatch(t, actualFields, expectedFields)
+						} else {
+							setIndex := slices.Index(actualFields, "--set")
+							setValuesIndex := setIndex + 1
+
+							actualSetFields := strings.Split(actualFields[setValuesIndex], ",")
+							expectedSetFields := strings.Split(expectedFields[setValuesIndex], ",")
+
+							assert.ElementsMatch(t, actualSetFields, expectedSetFields)
```

Comment:
This doesn't seem to be sufficient since we only compare the `--set` args and not the whole `template ` command.